pkg/importer/http-datasource.go

```diff
@@ -196,7 +197,19 @@ func (hs *HTTPDataSource) GetURL() *url.URL {
 
 // GetTerminationMessage returns data to be serialized and used as the termination message of the importer.
 func (hs *HTTPDataSource) GetTerminationMessage() *common.TerminationMessage {
-	return nil
+	if pullMethod, _ := util.ParseEnvVar(common.ImporterPullMethod, false); pullMethod != string(cdiv1.RegistryPullNode) {
+		return nil
+	}
+
+	info, err := getServerInfo(hs.ctx, fmt.Sprintf("%s://%s/info", hs.endpoint.Scheme, hs.endpoint.Host))
+	if err != nil {
+		klog.Errorf("%+v", err)
+		return nil
```

Comment:
Do we want to differentiate between fatal errors such as `no such endpoint` and other intermittent network errors that can be retried by restarting the pod?

Not insisting just have to raise this

Reply:
> How would it trigger the restart, exit with a code != 0?

That was my thinking, yes (you would just return an error here and that would exit with 1, IIRC)
But thinking more it's pretty complex to start differentiating the error types here so maybe leave as is

Reply:
Leaving as is for now.